webknossos/webknossos/dataset/dataset.py

```diff
@@ -247,6 +263,125 @@ def open(cls, dataset_path: Union[str, PathLike]) -> "Dataset":
+        * organization_id,
+        * sharing_token.
+        """
+        from webknossos.client.context import _get_context, webknossos_context
+
+        caller = inspect.stack()[1].function
+
+        match = re.match(_DATASET_URL_REGEX, dataset_name_or_url)
+        if match is not None:
+            assert (
+                organization_id is None
+                and sharing_token is None
+                and webknossos_url is None
+            ), (
+                f"When Dataset.{caller}() is be called with an annotation url, "
+                + f"e.g. Dataset.{caller}('https://webknossos.org/datasets/scalable_minds/l4_sample_dev/view'), "
+                + "organization_id, sharing_token and webknossos_url must not be set."
+            )
+            dataset_name = match.group("dataset_name")
+            organization_id = match.group("organization_id")
+            sharing_token = match.group("sharing_token")
+            webknossos_url = match.group("webknossos_url")
+        else:
+            dataset_name = dataset_name_or_url
+
+        current_context = _get_context()
+        if webknossos_url is not None and webknossos_url != current_context.url:
+            if sharing_token is None:
+                warnings.warn(
+                    f"The supplied url {webknossos_url} does not match your current context {current_context.url}. "
+                    + f"Using no token, only public datasets can used with Dataset.{caller}(). "
+                    + "Please see https://docs.webknossos.org/api/webknossos/client/context.html to adapt the URL and token."
+                )
+            assert organization_id is not None, (
+                f"Please supply the organization_id to Dataset.{caller}()."
+                f"The supplied url {webknossos_url} does not match your current context {current_context.url}. "
+                + "In this case organization_id can not be inferred."
+            )
+            context_manager: ContextManager[None] = webknossos_context(
+                webknossos_url, token=None
+            )
+        else:
+            if organization_id is None:
+                organization_id = current_context.organization_id
+
+            context_manager = nullcontext()
+
+        return context_manager, dataset_name, organization_id, sharing_token
+
+    @classmethod
+    def open_remote(
+        cls,
+        dataset_name_or_url: str,
+        organization_id: Optional[str] = None,
+        sharing_token: Optional[str] = None,
+        webknossos_url: Optional[str] = None,
+    ) -> "RemoteDataset":
+        """Opens a remote webknossos dataset. Image data is accessed via network requests.
+        Dataset metadata such as allowed teams or the sharing token can be read and set
+        via the respective `RemoteDataset` properties.
+
+        * `dataset_name_or_url` may be a dataset name or a full URL to a dataset view, e.g.
+          `https://webknossos.org/datasets/scalable_minds/l4_sample_dev/view`
```

Comment:
/view should be optional in the URL

Reply:
It is already, I added an example for this to the tests.